### PR TITLE
Case system improvements

### PR DESCRIPTION
**Pull request by corymcdonald**
#### Saved replies
Jul-15-2019 14-30-06
#### Shift click to assign multiple
Jul-15-2019 08-40-02

#### Features

- Add thumbnails to case 
- Fix link to case on publishers show page
- Mass assignment of cases
- Reload page when note is made internal

#### How To Test

Create a series of publishers and cases in `rails console`, here's a little script I made to do so
```ruby

first_names = %w(Emma Olivia Ava Isabella Sophia Charlotte Mia Amelia Harper Evelyn Liam Noah William James Oliver Benjamin Elijah Lucas Mason Owen)
last_names = %w(Smith Johnson Williams Jones Brown Davis Miller Wilson)

(1...100).each do |x|
  publisher = Publisher.create(email: "test#{x}@gmail.com", name: "#{first_names.sample} #{last_names.sample}")
  PublisherStatusUpdate.create!(publisher: publisher, status: 'active')
  PublisherStatusUpdate.create!(publisher: publisher, status: 'no_grants')

  @case = Case.create(publisher: publisher, 
    status: 'new', 
    solicit_question: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada metus erat, et cursus mi pretium eu. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Mauris tortor enim, dignissim quis est eget, tempor porttitor augue.',
    accident_question: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
  )
  @case.update(status: 'open')
end
```

Then you can work through the cases

### CODE REVIEW

File: app/controllers/admin/case_replies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+module Admin
+  class CaseRepliesController < AdminController
+    def index
+      @open_cases = Case.where(status: Case::OPEN)
```
Comment (yachtcaptain23) [inline app/controllers/admin/case_replies_controller.rb]: Deadcode?

File: app/controllers/admin/case_replies_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    end
+
+    def create
+      CaseReply.create(reply_params)
+      redirect_to admin_case_replies_path, flash: { notice: "Your saved reply was created successfully."}
+    end
+
+    def edit
+      @reply = CaseReply.find(params[:id])
+      @open_cases = Case.where(status: Case::OPEN)
```
Comment (yachtcaptain23) [inline app/controllers/admin/case_replies_controller.rb]: Deadcode?

File: app/controllers/admin/case_replies_controller.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      @assigned_cases = Case.where(assignee: current_user, status: Case::IN_PROGRESS)
+    end
+
+    def update
+      CaseReply.update(reply_params)
```
Comment (yachtcaptain23) [inline app/controllers/admin/case_replies_controller.rb]: This seems to update every CaseReply (which happened for me)

File: app/controllers/admin/case_replies_controller.rb (hunk 1 above)
Comment (corymcdonald) [reply app/controllers/admin/case_replies_controller.rb]: Used in the sidebar

File: app/controllers/admin/case_replies_controller.rb (hunk 2 above)
Comment (corymcdonald) [reply app/controllers/admin/case_replies_controller.rb]: Used in the sidebar

File: app/controllers/admin/case_replies_controller.rb (hunk 3 above)
Comment (corymcdonald) [reply app/controllers/admin/case_replies_controller.rb]: Whoops, forgot a find 
https://github.com/brave-intl/publishers/pull/2054/commits/1bf5f4c9616e662e38bb12bf8a677c5b761abfd4

File: app/javascript/packs/admin_case.js [hunk 4]
```diff
@@ -1,5 +1,65 @@
...
+        // store as last item clicked
+        lastItem = Array.prototype.indexOf.call(items, item);
+      } else {
+        // store as first item clicked
+        firstItem = Array.prototype.indexOf.call(items, item);
+        // unset last item
+        lastItem = null;
+      }
+
+      // do magic
```
Comment (dlipeles) [inline app/javascript/packs/admin_case.js]: 🔮

**Pull request merged**